src/Compilers/CSharp/Portable/BoundTree/BoundPattern.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class BoundPattern
+    {
+        internal bool IsNegated(out BoundPattern innerPattern)
+        {
+            innerPattern = this;
+            bool negated = false;
+            while (innerPattern is BoundNegatedPattern negatedPattern)
+            {
+                negated = !negated;
+                innerPattern = negatedPattern.Negated;
```

Comment:
>innerPattern [](start = 16, length = 12)

I would expect us to set ```innerPattern``` to null at the end if we return false.

Reply:
I'd prefer to set a usable value in all cases. And with `is not not C`, we should set inner pattern, even though the return value is `false`.

---
In reply to: [530509675](https://github.com/dotnet/roslyn/pull/49369#discussion_r530509675) [](ancestors = 530509675)

Reply:
> I'd prefer to set a usable value in all cases. 

Consider clearly documenting this behavior in a doc comment.

---
In reply to: [530597572](https://github.com/dotnet/roslyn/pull/49369#discussion_r530597572) [](ancestors = 530597572,530509675)